src/Aspire.Hosting/ApplicationModel/AllocatedEndpoint.cs

```diff
@@ -65,6 +67,11 @@ public AllocatedEndpoint(EndpointAnnotation endpoint, string address, int port,
     /// </summary>
     public string UriString => $"{UriScheme}://{EndPointString}";
 
+    /// <summary>
+    /// The associated service name created in DCP for this endpoint.
+    /// </summary>
+    public string? DcpServiceName { get; }
```

Comment:
I'm not sure this is a great public name in our application model.

Users don't really know (and shouldn't know IMO) about DCP at all. Is there a better name we can use here?

Reply:
`OrchestratorServiceName`?

Reply:
Can we make it internal for now? We can always make it public later.

Reply:
That makes it impossible to write a test. This is difficult because we need something that can appear in string interpolated expression. It doesn’t need to be the dcp expression but it has to be something that we can detect and replace with the dcp expression, at runtime .

There’s another alternative that comes to mind:

Make it internal:

Don’t evaluate this object inside of the IValueProvider implementation, but have that object flow all the way out to evaluation phase (GetValue) in ApplicationExecutor.

I think that means we’d need to manually process ReferenceExpressions in GetValue, handling target port specially.